coalib/bears/Bear.py

```diff
@@ -249,8 +250,19 @@ def execute(self, *args, **kwargs):
             result = self.run_bear_from_section(args, kwargs)
             return [] if result is None else list(result)
         except (Exception, SystemExit):
-            self.warn('Bear {} failed to run. Take a look at debug messages'
-                      ' (`-V`) for further information.'.format(name))
+            try:
+                if self.kind() == BEAR_KIND.GLOBAL:
+                    self.warn('Bear {} failed to run. Take a look '
+                              'at debug messages (`-V`) for further '
+                              'information.'.format(name))
+                else:
+                    self.warn('Bear {} failed to run on file {}. Take a look '
+                              'at debug messages (`-V`) for further '
+                              'information.'.format(name, args[0]))
+            except (Exception, SystemExit):
+                self.warn('Bear {} failed to run. Take a look '
+                          'at debug messages (`-V`) for further '
+                          'information.'.format(name))
```

Comment:
I would completely omit the try-catch^^

(Btw: Throwing an exception inside an except-block keeps the traceback alive, so don't `print`, but `raise` again :D)